### PR TITLE
Uniqueness constraint to find bugs faster.

### PR DESCRIPTION
**Pull request by prescod**
I had a bug in my code and it manifested as a Salesforce uniqueness constraint violation. Working backwards I realized that it was really a problem of the same bundle being processed twice without going through any kind of reset/requeue code. So I wanted to add this constraint to help catch these kinds of bugs faster.

I checked prod to see if it would break anything and on prod, only one bundle would violate this unqiueness constraint and surprise, surprise, it has a mysterious error about a Salesforce constraint violation.

https://sfdoc.sfdc.sh/publish/bundles/158/

So I think that this constraint will reduce mysterious downstream bugs.

To deploy to prod we'd just need to delete bundle 158. Staging doesn't seem to have any duplicate articles in bundles.

### CODE REVIEW

File: sfdoc/publish/models.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 class Article(models.Model):
     """Tracks created/updated articles per bundle."""
+    class Meta:
+        unique_together = [["bundle", "url_name"]]
```
Comment (davisagli) [inline sfdoc/publish/models.py]: wouldn't it be valid for url_name to be duplicated if there was previously another article at the same URL that was deleted?

Comment (prescod) [reply sfdoc/publish/models.py]: You're right! It's really the filename which is the "unique key" per bundle.

Comment (prescod) [reply sfdoc/publish/models.py]: Okay I had to re-read the code to convince myself but I think my initial understanding was correct.

An article is "deleted" if it exists on PowerOfUs at url_name and no matching url_name exists in the HTML bundle on disk.

In other words: the url_name is actually the unique identifier for articles. For a moment I thought that it was filenames but that was images I was thinking about.

Comment (davisagli) [reply sfdoc/publish/models.py]: Okay, I get it now.

**Comment by davisagli** [comment]
I'll approve this once someone takes responsibility for removing the duplicate in production.

**Comment by prescod** [comment]
image

**Comment by prescod** [comment]
$ heroku psql -a sfdoc

sfdoc::DATABASE=> select url_name, bundle_id, count(*) from publish_article group by url_name, bundle_id having count(*)>1;
 url_name | bundle_id | count 
----------+-----------+-------
(0 rows)

File: sfdoc/publish/models.py (hunk 1 above)
Comment (dextermilo) [reply sfdoc/publish/models.py]: @prescod you are correct there should be no case where a 'deleted' (archived) article has a url_name matching a url_name in the html from the bundle.

**Pull request merged**